### ISSUE TITLE
Annotation is not working properly

### ISSUE THREAD

**Issue by mpkumar87** [cornerstonejs/cornerstoneTools#554]
HI,
I am checking cornerstoneTool package in local and tested examples. In Annotation example, it create first annotation with text displaying below the line. But when try to create another annotation and text entered is not displayed for new annotation draw, instead it update the value of first annotation line. Please guide me to solve this problem.

Thanks.

**Comment by dannyrb** [comment]
Hello @mpkumar87, please include a screenshot and code example? Bonus points for an example on codepen that demonstrates the issue.

**Comment by mpkumar87** [comment]
Thanks for ur reply.. Example which is in github also not working for annotation.

https://rawgit.com/cornerstonejs/cornerstoneTools/master/examples/annotation/index.html

First annotation mark display value which i entered in dialog box. But further annotation mark not displaying value, it is getting update first annotation mark.

Thanks.

**Comment by swederik** [comment]
Looks like you're right. I wonder when that broke...

**Comment by galelis** [comment]
I have created a PR to fix this issue
https://github.com/cornerstonejs/cornerstoneTools/pull/557

@mpkumar87 Hey, I have analysed the issue and it was actually an issue only with example code that was not removing correctly the dialog events, so it was being called as many times as we have arrowAnnotations in the app.

**Comment by param88singh** [comment]
Helle swederik,

First of all thanks for your great effort for medical software(Dicom Viewer). I configured your latest versions of  cornerstone library. cornerstone-wado-image-loader - 2.0.0, cornerstone-tools - 2.3.3,
Here i seen annotation is not working accurately. It is almost difference of hundreds in length. But when i tried with yous oldest library(Augast 2017 versions). It is working very fine. Let me know what is the bug in your latest library. It is working good for almost all dicom files. But there is some dicom on which not working fine. I have attached the Example dicom file of DX modality. Check and confirm..
[DicomAnnotationIssue.zip](https://github.com/cornerstonejs/cornerstoneTools/files/2501313/DicomAnnotationIssue.zip)
